pkg/stargate/deployments.go

```diff
@@ -201,6 +200,20 @@ func NewDeployments(stargate *api.Stargate, dc *cassdcapi.CassandraDatacenter) m
 			deployment.Spec.Template.Labels[coreapi.K8ssandraClusterNameLabel] = klusterName
 			deployment.Spec.Template.Labels[coreapi.K8ssandraClusterNamespaceLabel] = klusterNamespace
 		}
+
+		if stargate.Spec.IsAuthEnabled() {
+			// Stargate reacts to the sole presence of this variable, regardless of its contents.
+			// When this variable is absent, Stargate will use AllowAllAuthenticator.
+			// When this variable is present, Stargate will by default use PasswordAuthenticator, unless overridden
+			// by the stargate.authenticator_class_name system property (see below, computeJvmOptions).
+			// Note that any other authenticator than PasswordAuthenticator will cause the REST APIs to be unusable,
+			// however the CQL API will still be usable.
+			deployment.Spec.Template.Spec.Containers[0].Env = append(
+				deployment.Spec.Template.Spec.Containers[0].Env,
+				corev1.EnvVar{Name: "ENABLE_AUTH", Value: "true"},
```

Comment:
Doesn't this env var enable Stargate's token based auth for its api requests? At least in earlier versions of Stargate when we did the initial integration with k8ssandra, we learned that it always had to be enabled regardless of whether Cassandra auth was enabled.

Reply:
I believe this isn't the case anymore. `ENABLE_AUTH` will trigger this:

https://github.com/stargate/stargate/blob/136f88bbf7417706043f045ec2ccb9331f4b0f75/persistence-cassandra-4.0/src/main/java/io/stargate/db/cassandra/Cassandra40PersistenceActivator.java#L109-L116

So basically it sets the authenticator to PasswordAuthenticator. Afaict it has no implications in the token based auth for the REST API.

Reply:
I stand corrected. I looked at Stargate code and confirmed that `ENABLE_AUTH` does indeed configure PasswordAuthenticator just as the comment here describes. 

Is the toke based api auth always enabled?

Reply:
Yes, you cannot disable it. And it is table-based by default, unless you configure a JWT server.